main_test.go

```diff
@@ -193,32 +198,85 @@ func runTest(t *testing.T, test *mainTest) {
+		name:         test.name,
+		input:        input,
+		eKind:        eKind,
+		stringOutput: strings.HasSuffix(test.golden, "_string_output.golden"),
+		extVars:      test.meta.extVars,
+		extCode:      test.meta.extCode,
 	})
 
-	// TODO(sbarzowski) report which files were updated
-	if *update {
-		err := ioutil.WriteFile(test.golden, []byte(result.output), 0666)
-		if err != nil {
-			t.Errorf("error updating golden files: %v", err)
+	compareGolden := func(inputPath, goldenPath, result string, golden []byte) {
+		if bytes.Compare(golden, []byte(result)) != 0 {
+			// TODO(sbarzowski) better reporting of differences in whitespace
+			// missing newline issues can be very subtle now
+			t.Fail()
+			t.Errorf("Mismatch when running %s. Golden: %s\n", inputPath, goldenPath)
+			data := diff(result, string(golden))
+			t.Errorf("diff %s jsonnet %s\n", goldenPath, inputPath)
+			t.Errorf(string(data))
 		}
-		return
 	}
-	golden := read(test.golden)
-	if bytes.Compare(golden, []byte(result.output)) != 0 {
-		// TODO(sbarzowski) better reporting of differences in whitespace
-		// missing newline issues can be very subtle now
-		t.Fail()
-		t.Errorf("Mismatch when running %s.jsonnet. Golden: %s\n", test.name, test.golden)
-		data := diff(result.output, string(golden))
-		t.Errorf("diff %s jsonnet %s.jsonnet\n", test.golden, test.name)
-		t.Errorf(string(data))
 
+	switch eKind {
+	default:
+		// TODO(sbarzowski) report which files were updated
+		if *update {
+			err := ioutil.WriteFile(test.golden, []byte(result.output), 0666)
+			if err != nil {
+				t.Errorf("error updating golden files: %v", err)
+			}
+			return
+		}
+		golden := read(test.golden)
+		compareGolden(test.input, test.golden, result.output, golden)
+	case evalKindMulti:
+		expectFiles, err := ioutil.ReadDir(test.golden)
+		if err != nil {
+			t.Fatalf("ReadDir(%q): %v", test.golden, err)
+		}
+		// TODO(sbarzowski) report which files were updated
+		if *update {
+			for fn, content := range result.outputMulti {
+				err := ioutil.WriteFile(filepath.Join(test.golden, fn), []byte(content), 0666)
+				if err != nil {
+					t.Errorf("error updating golden file %v: %v", fn, err)
+				}
+			}
+			// Delete excess files
+			for _, f := range expectFiles {
+				if _, ok := result.outputMulti[f.Name()]; ok {
+					continue
+				}
+				if err := os.Remove(filepath.Join(test.golden, f.Name())); err != nil {
+					t.Errorf("error removing golden file %v: %v", f.Name(), err)
+				}
+			}
+			return
+		}
+		goldenContent := map[string][]byte{}
+		for _, f := range expectFiles {
+			goldenContent[f.Name()] = read(filepath.Join(test.golden, f.Name()))
+		}
+		for fn, content := range result.outputMulti {
+			if _, ok := goldenContent[fn]; !ok {
+				t.Errorf("jsonnet outputted file %v which does not exist in goldens", fn)
+				continue
+			}
+			compareGolden(test.input, fn, content, goldenContent[fn])
+			delete(goldenContent, fn)
+		}
+		for fn := range goldenContent {
```

Comment:
Nit:  it would be more explicit, if instead of deleting elements from goldenContent above, we just checked in this loop if fn exists in result.outputMulti.

Reply:
Done.